### PR TITLE
Load capabilities for valid token through middleware

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Revolyssup** [comment]
@Abhishek-kumar09  This will solve your CI problems.

**Comment by Revolyssup** [comment]
@leecalcote  This PR makes sure that on subsequent Meshery server restarts, you dont have to login back again in order to see remote provider capabilities like MeshMap. This helps to unstuck the Cypress tests for meshery-extensions.

**Comment by leecalcote** [comment]
@Revolyssup do you mind knocking out a couple of the lint checks while you're in here (not directly related to your code change).

File: handlers/middlewares.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ func (h *Handler) ProviderMiddleware(next http.Handler) http.Handler {
 	}
 	return http.HandlerFunc(fn)
 }
+func (h *Handler) TokenHandlerMiddleware(next func(http.ResponseWriter, *http.Request, *models.Preference, *models.User, models.Provider)) func(http.ResponseWriter, *http.Request, *models.Preference, *models.User, models.Provider) {
```
Comment (leecalcote) [inline handlers/middlewares.go:44]: Will you kindly add a description of this function?

File: models/remote_provider.go [hunk 2]
```diff
@@ -127,12 +127,14 @@ func (l *RemoteProvider) loadCapabilities(token string) {
 	}
 }
 
+var gmc sync.Mutex
```
Comment (leecalcote) [inline models/remote_provider.go:130]: What does `gmc` stand for? "Global ... "

**Comment by leecalcote** [comment]
> @Revolyssup do you mind knocking out a couple of the lint checks while you're in here (not directly related to your code change).

Actually, @gr455 might have taken care of a couple of these already. 👏

File: handlers/middlewares.go (hunk 1 above)
Comment (leecalcote) [reply handlers/middlewares.go:44]: Reminder

File: models/remote_provider.go (hunk 2 above)
Comment (leecalcote) [reply models/remote_provider.go:130]: Reminder

**Comment by acald-creator** [comment]
bump,

is this PR still needed?

**Pull request closed without merging**